src/components/CheckboxRadioSwitch/CheckboxRadioSwitch.vue

```diff
@@ -210,6 +235,38 @@ export default {
 			validator: type => type === TYPE_CHECKBOX || type === TYPE_RADIO || type === TYPE_SWITCH,
 		},
 
+		/**
+		 * Toggle the alternative button style
+		 */
+		buttonVariant: {
+			type: Boolean,
+			default: false,
+		},
+
+		/**
+		 * Toggle the rounded corners for the alternative button style
+		 */
+		buttonVariantRounded: {
+			type: Boolean,
+			default: false,
+		},
+
+		/**
+		 * Toggle the top left and right rounded corners for the alternative button style
+		 */
+		buttonVariantFirst: {
+			type: Boolean,
+			default: false,
+		},
+
+		/**
+		 * Toggle the bottom left and right rounded corners for the alternative button style
+		 */
+		buttonVariantLast: {
+			type: Boolean,
+			default: false,
+		},
```

Comment:
This should be automatically calculated.
Grouped-by-name radio can easily be targeted with first and last :)

Reply:
Alright, now using `:first-of-type` and `:last-of-type` which will only work if they are siblings.

We still need to get the information about whether they are direct successors or not (grouped or not). Still using a prop for that.

I also added the ability to group vertically (video is up-to-date in the first comment).

Reply:
> We still need to get the information about whether they are direct successors or not (grouped or not). Still using a prop for that.

I don't think that's necessary (or even helpful). Grouped checkboxes need to be direct siblings with no checkboxes as siblings that are not grouped. So they need to be wrapped in a common container. Otherwise the markup will fail, without or with the grouped prop.

See this for example:
```vue
<template>
<div>
	<div>
	First toggle with three options
		<CheckboxRadioSwitch :checked.sync="sharingPermission" value="r" name="sharing_permission_radio" type="radio" :button-variant="true" :button-variant-v-grouped="true">Default permission read</CheckboxRadioSwitch>
		<CheckboxRadioSwitch :checked.sync="sharingPermission" value="rw" name="sharing_permission_radio" type="radio" :button-variant="true" :button-variant-v-grouped="true">Default permission read+write</CheckboxRadioSwitch>
		<CheckboxRadioSwitch :checked.sync="sharingPermission" value="rw" name="sharing_permission_radio" type="radio" :button-variant="true" :button-variant-v-grouped="true">Default permission read+write</CheckboxRadioSwitch>
	</div>
	<div>
	Second toggle with two options
		<CheckboxRadioSwitch :checked.sync="sharingPermission" value="r" name="sharing_permission_radio" type="radio" :button-variant="true" :button-variant-v-grouped="true">Default permission read</CheckboxRadioSwitch>
		<CheckboxRadioSwitch :checked.sync="sharingPermission" value="rw" name="sharing_permission_radio" type="radio" :button-variant="true" :button-variant-v-grouped="true">Default permission read+write</CheckboxRadioSwitch>
		Other toggle not grouped
		<CheckboxRadioSwitch :checked.sync="sharingPermission" value="rw" name="sharing_permission_radio" type="radio" :button-variant="true">Default permission read+write</CheckboxRadioSwitch>
		<br>
		sharingPermission: {{ sharingPermission }}
	</div>
</div>
</template>
<script>
export default {
	data() {
		return {
			sharingPermission: 'r',
		}
	}
}
</script>
```

Which gives:
![Screenshot 2022-06-20 at 12-18-46 Nextcloud Vue Style Guide](https://user-images.githubusercontent.com/2496460/174581088-55f647f9-9b45-4536-9719-a1f3504cb64b.png)

Notice that the corners of the second last checkbox are missing, since there is another one not belonging to the group afterwards. For the three grouped checkboxes wrapped in a container, it works fine.

So imho we only need a prop to indicate horizontal/vertical grouping (would be nice to figure that out by CSS as well, but I don't know how) and leave the rest to the proper markup (give a common container to the grouped boxes).